tests/test_wrappers.py

```diff
@@ -270,6 +270,31 @@ def failing_application(request):
     assert resp.status_code == 400
 
 
+def test_request_access_control():
+    request = wrappers.Request.from_values(
+        headers={
+            "Origin": "https://palletsprojects.com",
+            "Access-Control-Request-Headers": "X-A, X-B",
+            "Access-Control-Request-Method": "PUT",
+        },
+    )
+    assert request.access_control_origin == "https://palletsprojects.com"
+    assert request.access_control_request_headers == {"X-A", "X-B"}
+    assert request.access_control_request_method == "PUT"
+
+
+def test_response_access_control():
+    response = wrappers.Response("Hello World")
+    assert response.allow_credentials is None
+    response.allow_credentials = True
+    response.allow_headers = {"X-A", "X-B"}
```

Comment:
Let's use a list here, passing sets isn't very well defined (looks like we're implying order here, which would affect caching).

Reply:
Done.